### ISSUE TITLE
[bug] Error within useReadContract when passing Signer object as signerOrProvider

### ISSUE THREAD

**Issue by aramalipoor** [tmm/wagmi#367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.3.0

### Current Behavior

Source:
```ts
  const result = useContractReadWagmi(
    {
      addressOrName: contractAddress,
      contractInterface: contract.artifact.abi,
      signerOrProvider, // <-- when this is not provided there's no issue.
    },
    functionName as string,
    {
      args,
    }
  );
```

When passing "signerOrProvider" to `useReadContract()` looks like hashQueryKey cannot stringify properly. 

Not sure if it is intentional to allow `signerOrProvider` in here https://github.com/tmm/wagmi/blob/96e54382b117d51bcae6c4b85c547f3a1b3596f3/packages/react/src/hooks/contracts/useContractRead.ts#L84 🤔 

> TypeError: Converting circular structure to JSON --> starting at object with constructor 'f' | property '_readableState' -> object with constructor 'y' | property 'pipes' -> object with constructor 'i' | property '_readableState' -> object with constructor 'y' --- property 'pipes' closes the circle
![image](https://user-images.githubusercontent.com/1164589/165511910-8ae188e1-c507-4571-a20d-c4a6b48a8bd7.png)

### Expected Behavior

To create the query key with no issues.

### Steps To Reproduce

1. Get signer from useSigner()
2. Make a call to `useContractRead()` and pass `signerOrProvider`
3. Throws an error regarding circular structure

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://codesandbox.io/s/wagmi-circular-signer-error-94pg8o?file=/src/App.tsx

### Anything else?

Actually I'm not fully sure if it's a good idea to pass `signer` to read hooks at all, or always stick to `provider`. Mainly asking for example for cases where we want to call a "view" method which checks msg.sender.

My current understanding is:
- To provide "signer" whenever it's available
- Fallback to `undefined` when signer not available, so that useReadContract internally uses the correct provider.

**Comment by tmm** [comment]
Good find! Looking into it.

**Comment by peduarte** [comment]
Has this been published? I'm in `0.3.5` and I get the same error when using `signerOrProvider`

**Comment by peduarte** [comment]
i had a contractConfig object like this:
```js
// 📄 Contract
const contractConfig = {
  addressOrName: '0x...',
  contractInterface: contractInterface,
  signerOrProvider: signer
};
```

which I was using to in useContractRead like this:
```js
useContractRead(contractConfig, 'mintRate')
```

And that was throwing the error. But I’m also using `useContract`, which requires a signer. 

So to fix it, I split the contract config in two:
```js
  // 📄 Contract Config
  const contractReadConfig = {
    addressOrName: '0x62fd7401277B6d9be6c7AE995149A983bD6f1110',
    contractInterface: contractInterface
  };

  const contractConfig = {
    ...contractReadConfig,
    signerOrProvider: signer,
  };
```
